virtual_rainforest/core/axes.py

```diff
@@ -0,0 +1,551 @@
+            value.coords
+        )
+
+    def run_validation(self, value: DataArray, grid: Grid, **kwargs: Any) -> DataArray:
+        """Run validation on the inputs.
+
+        Validation will fail when the ``x`` and ``y`` coordinates:
+
+        * Fall on cell geometry boundaries: unambiguous coordinates within the geometry,
+          such as cell centroids, should be used.
+        * Are not at the same resolution, such that either multiple data cells map onto
+          a single grid cell or that some grid cells do not have matching data cells.
+        * Have too small an extent, so that again some grid cells do not have matching
+          data coordinates.
+
+        Args:
+            value: An input DataArray to check
+            grid: A Grid object giving the spatial configuration of the simulation.
+            kwargs: Other configuration details to be used.
+
+        Raises:
+            ValueError: when ``x`` and ``y`` values are not congruent with the ``Grid``.
+
+        Returns:
+            A DataArray with the ``x`` and ``y`` dimensions remapped onto the internal
+            ``cell_id`` dimension used in the :mod:`~virtual_rainforest.core.grid`
+            module.
+        """
+        # Get x and y coords to check the extents and cell coverage.
+        #
+        # TODO - Note that mapping all the cells here is a bit extreme with a square
+        # grid - could just map one row and column to confirm the indexing into the
+        # data, but this is a more general solution.
+
+        # Use .stack() to convert the axis into stacked pairs of values
+        xy_pairs = value.stack(cell=("y", "x")).coords
+        idx_pairs = value.drop_vars(("x", "y")).stack(cell=("y", "x")).coords
+
+        # Get the mapping of points onto the grid
+        idx_x, idx_y = grid.map_xy_to_cell_indexing(
+            x_coords=xy_pairs["x"].values,
+            y_coords=xy_pairs["y"].values,
+            x_idx=idx_pairs["x"].values,
+            y_idx=idx_pairs["y"].values,
+            strict=False,
+        )
+
+        # TODO - fine a way to enable strict = True - probably just kwargs.
+
+        # Now remap the grids from xy to cell_id - this uses the rather under described
+        # vectorized indexing feature in xarray:
+        #
+        #   https://docs.xarray.dev/en/stable/user-guide/indexing.html
+        #
+        # Specifically using DataArray.isel to select indexed dimensions by name, which
+        # avoids issues with different permutations of the axes - and map XY onto a
+        # common cell_id
+
+        return value.isel(
+            x=DataArray(idx_x, dims=["cell_id"]), y=DataArray(idx_y, dims=["cell_id"])
+        )
+
+
+class Spat_XY_Dim_Square(AxisValidator):
+    """Validate *x* and *y* dimensions on the *spatial* core axis on a square grid.
+
+    Applies to:
+        An input DataArray with ``x`` and ``y`` dimensions is assumed to provide a
```

Comment:
Again feel that the different data set this is applied to could be spelt out more explicitly

Reply:
Like the "Validation will fail if" sections are very helpful, and I can infer the required structure of the data set from there, but don't find it very clear here